File: cdm/cpp/scenario/SECondition.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "scenario/SECondition.h"
#include "substance/SESubstanceManager.h"
#include "patient/conditions/SEPatientCondition.h"
#include "system/environment/conditions/SEEnvironmentCondition.h"

SECondition::SECondition() : Loggable()
{
  m_Comment="";
}

SECondition::~SECondition()
{
  Clear();
}

void SECondition::Clear()
{
  m_Comment = "";
}

std::string SECondition::GetComment() const
{
  return m_Comment;
}
void SECondition::SetComment(const std::string& comment)
{
  m_Comment = comment;
}
bool SECondition::HasComment() const
{
  return m_Comment.empty()?false:true;
}
void SECondition::InvalidateComment()
{
  m_Comment = "";
}

File: cdm/csharp/engine/SECondition.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

public abstract class SECondition
{
  protected string comment;

  public SECondition()
  {
    comment = null;
  }

  public void Copy(SECondition other)
  {
    Clear();
    comment = other.comment;
  }

  public virtual void Clear()
  {
    comment = null;
  }

  public bool HasComment()
  {
    return !string.IsNullOrEmpty(comment);
  }
  public string GetComment()
  {
    return this.comment;
  }
  public void SetComment(string comment)
  {
    this.comment = comment;
  }
  public void InvalidateComment()
  {
    this.comment = null;
  }

  public abstract bool IsValid();
}
